src/extension.ts

```diff
@@ -369,19 +369,31 @@ export class Extension implements RunHooks {
     const projectPrefix = project.name ? `${project.name} — ` : '';
     const keyPrefix = configFile + ':' + project.name;
     let runProfile = existingProfiles.get(keyPrefix + ':run');
-    const projectTag = this._testTree.projectTag(project);
     const isDefault = false;
     const supportsContinuousRun = this._settingsModel.allowWatchingFiles.get();
     if (!runProfile)
-      runProfile = this._testController.createRunProfile(`${projectPrefix}${folderName}${path.sep}${configName}`, this._vscode.TestRunProfileKind.Run, this._scheduleTestRunRequest.bind(this, configFile, project.name, false), isDefault, projectTag, supportsContinuousRun);
+      runProfile = this._testController.createRunProfile(`${projectPrefix}${folderName}${path.sep}${configName}`, this._vscode.TestRunProfileKind.Run, this._scheduleTestRunRequest.bind(this, configFile, project.name, false), isDefault, undefined, supportsContinuousRun);
     this._runProfiles.set(keyPrefix + ':run', runProfile);
     let debugProfile = existingProfiles.get(keyPrefix + ':debug');
     if (!debugProfile)
-      debugProfile = this._testController.createRunProfile(`${projectPrefix}${folderName}${path.sep}${configName}`, this._vscode.TestRunProfileKind.Debug, this._scheduleTestRunRequest.bind(this, configFile, project.name, true), isDefault, projectTag, supportsContinuousRun);
+      debugProfile = this._testController.createRunProfile(`${projectPrefix}${folderName}${path.sep}${configName}`, this._vscode.TestRunProfileKind.Debug, this._scheduleTestRunRequest.bind(this, configFile, project.name, true), isDefault, undefined, supportsContinuousRun);
     this._runProfiles.set(keyPrefix + ':debug', debugProfile);
 
-    runProfile.onDidChangeDefault(enabled => project.model.setProjectEnabled(project, enabled));
-    debugProfile.onDidChangeDefault(enabled => project.model.setProjectEnabled(project, enabled));
+    // Run profile has the current isEnabled value as per vscode.
+    project.isEnabled = runProfile.isDefault;
```

Comment:
Shall we remove some `setTimeout(0)` logic in the vscode mock?

Reply:
It is still there in vscode.